dcmdump/src/dump.rs

```diff
@@ -0,0 +1,514 @@
+
+    meta_dump(&mut to, &meta, width)?;
+
+    println!("{:-<58}", "");
+
+    dump(&mut to, &obj, width, 0, no_text_limit)?;
+
+    Ok(())
+}
+
+pub fn dump_object(obj: &InMemDicomObject<StandardDataDictionary>, no_text_limit: bool) -> IoResult<()> {
+    let mut to = stdout();
+
+    let width = if let Some((width, _)) = term_size::dimensions() {
+        width as u32
+    } else {
+        120
+    };
+
+    println!("{:-<65}", "");
+
+    dump(&mut to, &obj, width, 0, no_text_limit)?;
+
+    Ok(())
+
+}
+
+#[inline]
+fn whitespace_or_null(c: char) -> bool {
+    c.is_whitespace() || c == '\0'
+}
+
+fn meta_dump<W>(to: &mut W, meta: &FileMetaTable, width: u32) -> IoResult<()>
+where
+    W: ?Sized + Write,
+{
+    writeln!(
+        to,
+        "{}: {}",
+        "Media Storage SOP Class UID".bold(),
+        meta.media_storage_sop_class_uid
+            .trim_end_matches(whitespace_or_null),
+    )?;
+    writeln!(
+        to,
+        "{}: {}",
+        "Media Storage SOP Instance UID".bold(),
+        meta.media_storage_sop_instance_uid
+            .trim_end_matches(whitespace_or_null),
+    )?;
+    if let Some(ts) = TransferSyntaxRegistry.get(&meta.transfer_syntax) {
+        writeln!(
+            to,
+            "{}: {} ({})",
+            "Transfer Syntax".bold(),
+            ts.uid(),
+            ts.name()
+        )?;
+    } else {
+        writeln!(
+            to,
+            "{}: {} («UNKNOWN»)",
+            "Transfer Syntax".bold(),
+            meta.transfer_syntax.trim_end_matches(whitespace_or_null)
+        )?;
+    }
+    writeln!(
+        to,
+        "{}: {}",
+        "Implementation Class UID".bold(),
+        meta.implementation_class_uid
+            .trim_end_matches(whitespace_or_null),
+    )?;
+
+    if let Some(v) = meta.implementation_version_name.as_ref() {
+        writeln!(to, "{}: {}", "Implementation version name".bold(), v.trim_end())?;
+    }
+
+    if let Some(v) = meta.source_application_entity_title.as_ref() {
+        writeln!(to, "{}: {}", "Source Application Entity Title".bold(), v.trim_end())?;
+    }
+
+    if let Some(v) = meta.sending_application_entity_title.as_ref() {
+        writeln!(to, "{}: {}", "Sending Application Entity Title".bold(), v.trim_end())?;
+    }
+
+    if let Some(v) = meta.receiving_application_entity_title.as_ref() {
+        writeln!(to, "{}: {}", "Receiving Application Entity Title".bold(), v.trim_end())?;
+    }
+
+    if let Some(v) = meta.private_information_creator_uid.as_ref() {
+        writeln!(
+            to,
+            "{}: {}",
+            "Private Information Creator UID".bold(),
+            v.trim_end_matches(whitespace_or_null)
+        )?;
+    }
+
+    if let Some(v) = meta.private_information.as_ref() {
+        writeln!(to, "{}: {}",
+            "Private Information".bold(),
+            format_value_list(v.iter().map(|n| format!("{:02X}", n)), width, false)
+        )?;
+    }
+
+    writeln!(to)?;
+    Ok(())
+}
+
+pub fn dump<W, D>(
```

Comment:
Oh, and now that they are part of a library, some documentation would be nice too.